### ISSUE TITLE
%{dirname:foo} is returning foo

### ISSUE THREAD

**Issue by mlschroe** [rpm-software-management/rpm#2928]
We're testing for it in the test cases so it might be intentional, but is there any good reason why `%{dirname:foo}` returns `foo` instead of `` or `.`? At least the documentation says it is a "dirname(1) macro analogue"...

**Comment by ffesti** [comment]
My suspicion is that %dirname was used there not to test for it returning the filename unchanged but it was chosen as leaving the parameter unchanged made it a good test case for undefined arguments.

**Comment by pmatilai** [comment]
Indeed, there's no deeper meaning to the test, it's all about argument handling and nothing to do with dirname itself (the context of c5909bd8698e56ffa4d1bb380116d61d6b48b8dd makes it quite clear)
